### PR TITLE
Benchmark harness

### PR DESCRIPTION
**Pull request by noahgibbs**
Okay. I've simply removed the regexp we can't compile. So we're down to two. But I'd like to get this merged so I can add to it in reasonable safety.

### CODE REVIEW

File: bench/run_benchmarks.rb [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    raise
+  end
+
+  [
+    [Regexp.new(source), :ruby, "Ruby built-in regexp"],
+    [re_basic, :re, "uncompiled RegularExpression object"],
+    [re_x86, :re_x86, "RegularExpression x86 compiler"],
+    [re_ruby, :re_ruby, "RegularExpression Ruby compiler"]
+  ].each do |match_obj, samples_name, matcher_description|
+    did_match = match_obj.match?(value)
```
Comment (kddnewton) [inline bench/run_benchmarks.rb:50]: This could probably be `!match_object.match?(value).nil?`

Comment (noahgibbs) [reply bench/run_benchmarks.rb:50]: I don't think so. A no-match returns false, not nil.

Comment (noahgibbs) [reply bench/run_benchmarks.rb:50]: Or at least, I think I've seen that. Let me go try it.

Comment (noahgibbs) [reply bench/run_benchmarks.rb:50]: Huh. Ruby built-in regexps are sometimes returning false for regexp match failures. So yeah, "nil?" doesn't work here :-(

**Pull request merged**